subsys/bluetooth/mesh/scheduler_srv.c

```diff
@@ -730,11 +746,17 @@ static void scheduler_srv_reset(struct bt_mesh_model *model)
 	struct bt_mesh_scheduler_srv *srv = model->user_data;
 
 	srv->idx = BT_MESH_SCHEDULER_ACTION_ENTRY_COUNT;
-	srv->status_bitmap = 0;
+	srv->active_bitmap = 0;
 	k_delayed_work_cancel(&srv->delayed_work);
 	net_buf_simple_reset(srv->pub.msg);
 	memset(&srv->sch_reg, 0, sizeof(srv->sch_reg));
 
+	for (int i = 0; i < BT_MESH_SCHEDULER_ACTION_ENTRY_COUNT; i++) {
+		srv->sched_tai[i].sec = 0;
+		srv->sched_tai[i].subsec = 0;
+		srv->sch_reg[i].action = BT_MESH_SCHEDULER_NO_ACTIONS;
+	}
```

Comment:
Is both this and the memset necessary? Couldn't we just set `.action` for all, as that's the decision point?

Reply:
I guess it is possible to not set zeroes in tai structure. The server will have garbage there until it is configured. Not nice from my point of view but does not impact anything.
However, I'm not sure it is a good idea to store garbage into flash if memset is removed.

Reply:
Could we just delete the unused entires from flash instead?

Reply:
Deletion will add yet another write to the entry and entry name and will make flash wear worse (name + value both are removed). It does not buy anything. Even if stack deletes entries now at some point they will get used anyway.

Reply:
deleted both in init and reset